src/main.tf

```diff
@@ -16,22 +16,12 @@ resource "azurerm_storage_account" "main" {
   account_tier                      = "Standard"
   account_kind                      = "StorageV2"
   access_tier                       = "Hot"
-  account_replication_type          = var.redundancy.replication_type
+  account_replication_type          = var.redundancy.data_protection ? var.redundancy.replication_type : "LRS"
   min_tls_version                   = "TLS1_2"
-  public_network_access_enabled     = false
+  public_network_access_enabled     = true
```

Comment:
For the keen-eyed, yes `public` is set to true. It's a weird firewall rule thing I don't yet fully understand with the managed service offerings of app-service and function-app. I tried all other permutations but this works and it is still blocked by RBAC. We can push this more private or not later IMO.

Reply:
Can we? what happens if we change this from public to private in everyone's bundle?

Reply:
That's a good point. I can check if that recreates the resource

Reply:
It's not marked that it would recreate. 
https://github.com/hashicorp/terraform-provider-azurerm/blob/7a3afb9d59a4cca6b1c410a3cd1a2d9db1d92d85/internal/services/storage/storage_account_resource.go#L350

Reply:
I think I have an instance up, let me cycle that real quick

Reply:
We're safe, good call! Let me add a comment to this real quick.

<img width="875" alt="Screenshot 2023-01-10 at 9 21 04 PM" src="https://user-images.githubusercontent.com/651833/211723905-486b0e55-1758-43e8-a870-9f5338c98edd.png">